numpy/core/src/umath/ufunc_object.c

```diff
@@ -2868,32 +2868,42 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
     if (identity == NULL) {
         return NULL;
     }
+
+    /* Get the initializer */
+    if (initializer == NULL) {
+        initializer = identity;
+    }
+    else {
+        Py_DECREF(identity);
```

Comment:
If we do as suggested above, the `Py_DECREF` can obviously be removed.

Reply:
See comment to above.